### PR TITLE
use cryptographic key pairs for encryption keys

### PR DESCRIPTION
**Pull request by efim-poberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Server.hs [hunk 1]
```diff
@@ -105,10 +106,10 @@ verifyTransmission (sig, t@(corrId, queueId, cmd)) = do
       st <- asks queueStore
       qr <- atomically $ getQueue st party queueId
       return $ either smpErr f qr
-    verifySend :: C.Signature -> Maybe C.PublicKey -> Cmd
+    verifySend :: C.Signature -> Maybe VerificationKey -> Cmd
     verifySend "" = maybe cmd (const authErr)
     verifySend _ = maybe authErr verifySignature
-    verifySignature :: C.PublicKey -> Cmd
+    verifySignature :: VerificationKey -> Cmd
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply src/Simplex/Messaging/Server.hs]: done

File: src/Simplex/Messaging/Server.hs [hunk 2]
```diff
@@ -105,10 +106,10 @@ verifyTransmission (sig, t@(corrId, queueId, cmd)) = do
       st <- asks queueStore
       qr <- atomically $ getQueue st party queueId
       return $ either smpErr f qr
-    verifySend :: C.Signature -> Maybe C.PublicKey -> Cmd
+    verifySend :: C.Signature -> Maybe VerificationKey -> Cmd
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply src/Simplex/Messaging/Server.hs]: done

File: src/Simplex/Messaging/Agent/Client.hs [hunk 3]
```diff
@@ -223,15 +220,15 @@ sendConfirmation c SendQueue {server, sndId} senderKey = do
       -- TODO encryption
       return msg
 
-sendHello :: forall m. AgentMonad m => AgentClient -> SendQueue -> m ()
-sendHello c SendQueue {server, sndId, sndPrivateKey, encryptKey} = do
-  msg <- mkHello "5678" $ AckMode On -- TODO verifyKey
+sendHello :: forall m. AgentMonad m => AgentClient -> SendQueue -> VerificationKey -> m ()
+sendHello c SendQueue {server, sndId, sndPrivateKey, encryptKey} verifyKey = do
+  msg <- mkHello verifyKey $ AckMode On
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply src/Simplex/Messaging/Agent/Client.hs]: done

**Review by epoberezkin — APPROVED** [review]

File: src/Simplex/Messaging/Agent/Store.hs [hunk 4]
```diff
@@ -23,19 +23,18 @@ import Data.Time.Clock (UTCTime)
...
   { server :: SMPServer,
     rcvId :: SMP.RecipientId,
     connAlias :: ConnAlias,
-    rcvPrivateKey :: C.PrivateKey,
+    rcvPrivateKey :: RecipientPrivateKey,
     sndId :: Maybe SMP.SenderId,
-    sndKey :: Maybe C.PublicKey,
-    decryptKey :: PrivateKey,
-    verifyKey :: Maybe PublicKey,
+    sndKey :: Maybe SenderPublicKey,
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply src/Simplex/Messaging/Agent/Store.hs:35]: although maybe not - it saved in order to re-send it to the server if the connection fails (although we don't have any logic to re-try)

[the remark this answers is not in the document]
Comment (efim-poberezkin) [reply src/Simplex/Messaging/Agent/Store.hs:35]: yes, I think it was initial reasoning

**Pull request merged**